jina/types/arrays/neural_ops.py

```diff
@@ -72,7 +72,7 @@ def match(
                 lhv = DocumentArray(lhv)
 
         rhv = darray
-        if traversal_rdarray:
```

Comment:
The solution should be in traverse to do the check and in flatten avoid flattening if only one sequence is passed